src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -97,12 +103,75 @@ private function completeDefinition($id, Definition $definition)
             $this->container->addResource(static::createResourceForClass($reflectionClass));
         }
 
-        if (!$constructor = $reflectionClass->getConstructor()) {
-            return;
+        $methodsCalled = array();
+        foreach ($definition->getMethodCalls() as $methodCall) {
+            $methodsCalled[$methodCall[0]] = true;
         }
 
-        $arguments = $definition->getArguments();
-        foreach ($constructor->getParameters() as $index => $parameter) {
+        foreach ($this->getMethodsToAutowire($id, $reflectionClass, $autowired) as $reflectionMethod) {
+            if (!isset($methodsCalled[$reflectionMethod->name])) {
+                $this->autowireMethod($id, $definition, $reflectionMethod);
+            }
+        }
+    }
+
+    /**
+     * Gets the list of methods to autowire.
+     *
+     * @param string           $id
+     * @param \ReflectionClass $reflectionClass
+     * @param string[]         $autowired
+     *
+     * @return \ReflectionMethod[]
+     */
+    private function getMethodsToAutowire($id, \ReflectionClass $reflectionClass, array $autowired)
+    {
+        $found = array();
+        $regexList = array();
+        foreach ($autowired as $pattern) {
+            $regexList[] = '/^'.str_replace('\*', '.*', preg_quote($pattern, '/')).'$/i';
```

Comment:
`*` comes from glob patterns, and there, `*` => `/.*/`, we shouldn't be the one redefining old semantics people are used to since years.